apps/engine/lib/execution/src/worker.rs

```diff
@@ -495,7 +495,7 @@ impl Worker {
         source: Language,
     ) -> Result<()> {
         if let Some(pending) = self.tasks.inner.get_mut(&msg.task_id) {
-            let next = WorkerHandler::handle_worker_message(&mut pending.task, msg.payload)?;
+            let next = WorkerHandler::handle_worker_message(&mut *pending.task, msg.payload)?;
```

Comment:
It is kinda ugly to have to put the `*` here. There might be a way around that, but I couldn't think of it.